docs/csharp/reference-semantics-with-value-types.md

```diff
@@ -0,0 +1,213 @@
+---
+title: Reference semantics with value types
+description: Understand the language features that minimize copying structures safely
+author: billwagner
+ms.author: wiwagn
+ms.date: 11/10/2017
+ms.topic: article
+ms.prod: .net
+ms.technology: devlang-csharp
+ms.devlang: csharp
+ms.custom: mvc
+---
+
+# Reference semantics with value types
+
+An advantage to using value types is that they often avoid heap allocations.
+The corresponding disadvantage is that they are copied by value. This tradeoff
+makes it harder to optimize algorithms that operate on large amounts of
+data. New language features in C# 7.2 provide mechanisms that enable pass-by-reference
+semantics with value types. If you use these features wisely you can minimize both allocations
+and copy operations. This article explores those new features.
+
+Much of the sample code in this article demonstrates features added in C# 7.2. In order to
+use those features, you have to configure your project to use C# 7.2 or later in your
+project. You can use Visual Studio to select it. For each project, select **Project** from
+the menu, then **Properties**. Select the **Build** tab and click **Advanced**. From there,
+you can configure the language version. Choose either "7.2", or "latest".  Or you can
+edit the *csproj* file and add the following node:
+
+```XML
+  <PropertyGroup>
+    <LangVersion>7.2</LangVersion>
+  </PropertyGroup>
+```
+
+You can use either "7.2" or "latest" for the value.
```

Comment:
Given that there are multiple ways of updating a project (UI settings, editing csproj, code fix), consider factoring this section to a dedicated page, and referenced from here.
Then you can just have one line: "In order to use those features, you have to configure your project to use C# 7.2 or later in your project, by setting the language version to "7.2" or "latest" per _these instructions_".

Reply:
I'd agree. 

I want to take a bit more time to do that, so I made an issue that can get addressed in this or the next sprint: #3685